crates/wasmparser/src/readers/core/types.rs

```diff
@@ -1165,6 +1174,18 @@ impl RefType {
         self.as_u32() & Self::CONCRETE_BIT != 0
     }
 
+    /// Is this a reference to a shared type?
+    ///
+    /// Note that we cannot know if a concrete type is shared unless we read the
+    /// type at the index, which is not possible here (i.e., `None`).
+    pub const fn is_shared(&self) -> Option<bool> {
+        if self.is_concrete_type_ref() {
+            None
+        } else {
+            Some(self.as_u32() & Self::SHARED_BIT != 0)
+        }
+    }
```

Comment:
Sure. I have tried this a couple of times and it's not as easy as it might seem. How about I remove the various levels of this and switch the logic over to a `WasmModuleResources` function in another PR?

Reply:
Done, see ad37d62.